core/src/main/java/org/apache/calcite/tools/RelBuilder.java

```diff
@@ -2497,6 +2509,17 @@ public RelBuilder sortLimit(int offset, int fetch, RexNode... nodes) {
     return sortLimit(offset, fetch, ImmutableList.copyOf(nodes));
   }
 
+  /** Creates a {@link Sort} by specifying collations.
+   */
+  public RelBuilder sort(List<RelFieldCollation> collations) {
+    final ImmutableList.Builder<RexNode> builder = ImmutableList.builder();
+    for (RelFieldCollation collation : collations) {
+      builder.add(collation.direction.isDescending()
+          ? desc(field(collation.getFieldIndex())) : field(collation.getFieldIndex()));
+    }
+    return sortLimit(-1, -1, builder.build());
```

Comment:
Instead of doing this, can we have a method to sort(RelCollation)? like `RelBuilder.exchange(RelDistribution distribution)`

Reply:
done.

Reply:
This will be easier:
```
final RelNode sort =
              struct.sortFactory.createSort(peek(), collation,
                  null, null);
          replaceTop(sort);
          return this;
```

Reply:
I try to reuse the existing logic as much as possible. Currently all the sort operator creations are through sortLimit().

Reply:
It looks like premature optimization. We pass to it a complete stuff, this method tear the stuff down and pass all the parts to `sortLimit`, and assemble it back. But leave it to you to decide.

Reply:
BTW, this will drop the null direction information of RelCollation unexpectedly.

Reply:
Missing null direction is a deal breaker. I now update it to use the approach you suggested.